### ISSUE TITLE
FallThrough check false positive with labelled break

### ISSUE THREAD

**Issue by sbj42** [checkstyle/checkstyle#11230]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#FallThrough
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```cmd
> javac Test.java
(success)

> type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="FallThrough">
    </module>
  </module>
</module>

> type Test.java
public class Test {
    public int test(int a, int b) {
        top: switch (a) {
            case 1:
                switch (b) {
                    case 10:
                        break top;
                    default:
                        return 11;
                }
            case 2: // line 11
                return 2;
        }
        return -1;
    }
}

> java -jar checkstyle-0.2.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] Test.java:11:13: Fall through from previous branch of the switch statement. [FallThrough]
Audit done.
Checkstyle ends with 1 errors.
```

---

Expected result is no errors.  This code never falls through from case 1 to case 2.  The problem seems to only happen with a labelled break.  If I use `return 11` instead of `break top` then checkstyle does not find a problem.

**Comment by sbj42** [comment]
!3885 is also about a FallThrough false positive, but it focuses on infinite loops, which are harder to detect.

**Comment by romani** [comment]
We should not violate break to label. Any break should be good to consider it as valid code.

**Comment by romani** [comment]
```
rivanov@p5510:/var/tmp$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
       <module name="NonEmptyAtclauseDescription"/>
    </module>

</module>

rivanov@p5510:/var/tmp$ cat Test.java 
public class Test {
    public int test(int a, int b) {
        top: switch (a) {
            case 1:
                break top;
            case 2: // line 11
                return 2;
        }
        return -1;
    }
}
rivanov@p5510:/var/tmp$ java -jar checkstyle-10.9.1-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

we need to behave same as in this case, nested switch shoudl not be something special

**Comment by rnveach** [comment]
@romani `NonEmptyAtclauseDescription` is a javadoc check. What does this have to do with FallThroughCheck which is for switches?

**Comment by romani** [comment]
fixed, same behavior.

**Comment by aayushRedHat** [comment]
@romani I am taking this
